java-frontend/src/main/java/org/sonar/java/se/constraint/ObjectConstraint.java

```diff
@@ -23,42 +23,60 @@
 import java.util.Objects;
 import java.util.function.Predicate;
 
-public class ObjectConstraint implements Constraint {
+public class ObjectConstraint<S extends ObjectConstraint.Status> implements Constraint {
 
-  public static final ObjectConstraint NOT_NULL = new ObjectConstraint(false, true, null);
+  public interface Status {
+    default String valueAsString() {
+      return "";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this method and declare a constant for this value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3400)